torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -292,6 +296,23 @@ at::Tensor inferAndAlloc(
         ") for the buffer ",
         tv->toString());
     inferred_sizes.push_back(inferred_val.value());
+    if (inferred_val.value() == 1 &&
+        expanded_map.count(expanded_sizes.size())) {
```

Comment:
nit: took me some time to figure out why the count argument is the size of the vector.

Reply:
Added a comment and some logic to check inferred size and expanded size are consistent.